sklearn/cluster/k_means_.py

```diff
@@ -1415,3 +1416,25 @@ def predict(self, X):
 
         X = self._check_test_data(X)
         return self._labels_inertia_minibatch(X)[0]
+   
+    def _reset_estimator(self):
```

Comment:
I don't think you need to revert back to the more readable form, but I've not investigated why you need to clear all these attributes anyway...